activerecord/lib/active_record/relation/calculations.rb

```diff
@@ -483,7 +483,7 @@ def execute_simple_calculation(operation, column_name, distinct) # :nodoc:
           # PostgreSQL doesn't like ORDER BY when there are no GROUP BY
           relation = unscope(:order).distinct!(false)
 
-          column = aggregate_column(column_name)
+          column = aggregate_column(column_name, relation)
```

Comment:
Can you make `aggregate_column` protected instead of `arel_column`?

```suggestion
          column = relation.aggregate_column(column_name)
```

Reply:
Done!